frontend/relay-workflows-lib/lib/components/TemplatesList.tsx

```diff
@@ -22,41 +23,78 @@ const templatesListQuery = graphql`
 
 export default function TemplatesList() {
   const data = useLazyLoadQuery<TemplatesListQuery>(templatesListQuery, {});
-  const templates = data.workflowTemplates.nodes;
+  const [search, setSearch] = useState("");
+  const [filteredTemplates, setFilteredTemplates] = useState(
+    data.workflowTemplates.nodes,
+  );
+
+  const handleSearch = (search: string) => {
+    setSearch(search);
+  };
 
   const {
     pageNumber,
     setPageNumber,
     totalPages,
     paginatedItems: paginatedPosts,
-  } = useClientSidePagination(templates, 10);
+  } = useClientSidePagination(filteredTemplates, 10);
 
   const handlePageChange = (_event: ChangeEvent<unknown>, page: number) => {
     setPageNumber(page);
   };
 
+  useEffect(() => {
```

Comment:
then instead of a `useEffect` here, you can use a `useMemo` e.g.
```
const filteredTemplates = useMemo(() => {
  const upperSearch = search.toUpperCase();
  return templates.filter(template => 
  ...
  }, [search, templates])
```